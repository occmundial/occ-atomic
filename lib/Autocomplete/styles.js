import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
