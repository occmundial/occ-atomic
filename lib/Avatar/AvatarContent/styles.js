import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
