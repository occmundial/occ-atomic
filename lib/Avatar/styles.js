import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
