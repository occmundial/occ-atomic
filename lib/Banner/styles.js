import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
