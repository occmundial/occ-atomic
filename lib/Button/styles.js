import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
