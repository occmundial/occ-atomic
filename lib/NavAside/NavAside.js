import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card';
import Flexbox from '../Flexbox';
import Icon from '../Icon';
import colors from '../subatomic/colors';

class NavAside extends React.Component {
	constructor(props) {
		super(props);
		this.onKeyDown = this.onKeyDown.bind(this);
	}

	componentDidMount() {
		window.addEventListener('keydown', this.onKeyDown);
	}

	componentWillUnmount() {
		window.removeEventListener('keydown', this.onKeyDown);
	}

	onKeyDown({ which }) {
		if (which == 27) {
			this.props.onClose();
		}
	}

	avoidClose(e) {
		e.stopPropagation();
	}

	render() {
		const {
			classes,
			top,
			children,
			onClose,
			show,
			width,
			onTransitionEnd,
			right
		} = this.props;
		return (
			<div
				className={`${
					show
						? ` ${classes.overlay} ${classes.overlayShow}`
						: ` ${classes.overlay}`
				}`}
				onClick={onClose}
				onTransitionEnd={onTransitionEnd}
			>
				<div
					className={`${
						show
							? right
								? `${classes.ShowElement} ${classes.OpenElementRight}`
								: `${classes.ShowElement} ${classes.OpenElement}`
							: `${classes.ShowElement}`
					}${
						width && right
							? ` ${classes.customeWidthRight}`
							: ` ${classes.customeWidth}`
					}${right ? ` ${classes.showRight}` : ''}`}
					onClick={this.avoidClose}
				>
					<Card shadow={5} className={`${classes.card}`}>
						<Flexbox
							display="flex"
							justifyContent="end"
							alignItems="start"
							className={classes.top}
						>
							{top && (
								<Flexbox flex="1" className={classes.topContent}>
									{top}
								</Flexbox>
							)}
							<div className={classes.closeIcon}>
								<Icon
									iconName="close"
									width={24}
									height={24}
									colors={[colors.grey900]}
									onClick={onClose}
									testId="navaside__close-icon"
								/>
							</div>
						</Flexbox>
						<div className={classes.content}>{children}</div>
					</Card>
				</div>
			</div>
		);
	}
}

NavAside.propTypes = {
	classes: PropTypes.object.isRequired,
	/** Will display any element in the top position */
	top: PropTypes.node,
	/** Will display any element in the bottom position */
	children: PropTypes.node,
	/** Custome width for aside */
	width: PropTypes.number,
	/** Custome function for closing aside */
	onClose: PropTypes.func,
	/** Boolean value to show or hide the Aside component. */
	show: PropTypes.bool,
	/** This function is used by an internal wrapper component to manage the hiding animation and unmounting properly. */
	onTransitionEnd: PropTypes.func,
	/** Boolean value to show the Aside component in the right side */
	right: PropTypes.bool
};

export default NavAside;
