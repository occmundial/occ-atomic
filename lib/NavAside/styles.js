import grid from '../subatomic/grid';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';

const ASIDE_SIZE = 320;

const styles = {
  '@global': {
    body: {
        overflow:'hidden'
    }
  },
  '@keyframes overlayFadeIn': {
    from: { opacity: 0 },
    to: { opacity: 1 }
  },
  '@keyframes showFromLeft': {
    from: {
      left: -ASIDE_SIZE
    },
    to: { left: 0 }
  },
  '@keyframes showFromRight': {
    from: {
      right: -ASIDE_SIZE
    },
    to: { right: 0 }
  },
  overlay: {
    position: 'fixed',
    width: '100%',
    height: '100%',
    background: colors.bg.overlay,
    top: 0,
    left: 0,
    zIndex: 1000,
    cursor: 'pointer',
    transition: '0.3s opacity'
  },
  overlayShow: {
    animation: '0.3s overlayFadeIn linear'
  },
  overlayHide: {
    opacity: 0
  },
  block: {
    height: '100%',
    width: ASIDE_SIZE,
    maxWidth: `calc(100% - ${spacing['size-6']})`,
    position: 'fixed',
    top: 0,
    backgroundColor: colors.bg.surface.default,
    overflow: 'hidden',
    transition: '0.3s left, 0.3s right'
  },
  showLeft: {
    animation: '0.3s showFromLeft ease-out',
    left: 0
  },
  showRight: {
    animation: '0.3s showFromRight ease-out',
    right: 0
  },
  hideLeft: {
    left: -ASIDE_SIZE
  },
  hideRight: {
    right: -ASIDE_SIZE
  },
  card: {
    cursor: 'auto',
    height: '100%',
    padding: 0,
    overflow: 'auto'
  },
  content: {
    backgroundColor: colors.bg.surface.default,
    padding: spacing['size-4'],
    [`@media screen and (min-width:${grid.sm}px)`]: {
      padding: spacing['size-6']
    }
  },
  top: {
    backgroundColor: colors.bg.default,
    padding: spacing['size-4'],
    [`@media screen and (min-width:${grid.sm}px)`]: {
      padding: spacing['size-6']
    }
  },
  topContent: {
    minWidth: 0
  },
  closeIcon: {
    height: spacing['size-6'],
    display: 'flex',
    alignItems: 'center'
  }
};

export default styles;
