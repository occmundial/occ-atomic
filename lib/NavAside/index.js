import React from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';
import injectSheet from 'react-jss';
import NavAsideComponent from './NavAside';
import styles from './styles';

const NavAside = injectSheet(styles)(NavAsideComponent);

class NavAsidePortal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            mount: false
        };
        this.onTransitionEnd = this.onTransitionEnd.bind(this);
    }

    componentWillReceiveProps(props) {
        if (props.show && !this.props.show) {
            this.setState({
                mount: true
            });
        }
    }

    onTransitionEnd() {
        this.setState({
            mount: false
        });
    }

    render() {
        const { mount } = this.state;
        const { container, show } = this.props;
        const NavAsideComponent = <NavAside {...this.props} onTransitionEnd={(!show && mount) ? this.onTransitionEnd : null} />;
        if (container) {
            return (
                mount ? ReactDOM.createPortal(
                    NavAsideComponent,
                    container
                ) : null
            );
        } else {
            return (
                mount ? (
                    NavAsideComponent
                ) : null
            );
        }
    }
}

NavAsidePortal.propTypes = {
    show: PropTypes.bool,
    container: PropTypes.object
};

export default NavAsidePortal;
