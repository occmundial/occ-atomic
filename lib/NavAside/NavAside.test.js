import React from 'react';
import { shallow, mount } from 'enzyme';
import NavAside from './NavAside';
import styles from './styles';
import NavAsidePortal from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("NavAside", () => {

    it('matches the snapshot', () => {
        const onClose = jest.fn();
        const wrapper = shallow(
            <NavAside classes={classes} onClose={onClose} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('calls the onClose function when clicking the overlay', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <NavAside classes={classes} onClose={onClose} />
        );
        wrapper.find('.overlay').simulate('click');
        expect(onClose.mock.calls.length).toBe(1);
    });

    it('toggles between show and hide', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <NavAside classes={classes} onClose={onClose} show={true} />
        );
        expect(wrapper.find('.overlayShow').length).toBe(1);
        wrapper.setProps({ show: false });
        expect(wrapper.find('.overlayShow').length).toBe(0);
    });

    it('renders the content', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <NavAside classes={classes} onClose={onClose}>
                Content
            </NavAside>
        );
        expect(wrapper.find('.content').at(0).text()).toBe('Content');
    });

    it('renders the top', () => {
      const onClose = jest.fn();
      const wrapper = mount(
          <NavAside classes={classes} onClose={onClose} top={'Content'} />
      );
      expect(wrapper.find('.top').at(0).text()).toBe('Content');
  });


});

describe("ModalPortal", () => {

    it('matches the snapshot', () => {
        const navAsideRoot = global.document.createElement('div');
        const wrapper = shallow(
            <NavAsidePortal container={navAsideRoot} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('mounts and unmounts the component', () => {
        const onClose = jest.fn();
        const navAsideRoot = global.document.createElement('div');
        navAsideRoot.setAttribute('id', 'navAside-root');
        const body = global.document.querySelector('body');
        body.appendChild(navAsideRoot);
        const wrapper = mount(
            <NavAsidePortal container={navAsideRoot} onClose={onClose} />
        );
        expect(navAsideRoot.hasChildNodes()).toBeFalsy();
        wrapper.setProps({ show: true });
        expect(navAsideRoot.hasChildNodes()).toBeTruthy();
        wrapper.setProps({ show: false });
        wrapper.setState({ mount: false });
        expect(navAsideRoot.hasChildNodes()).toBeFalsy();
    });

});

describe("NavAside styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});