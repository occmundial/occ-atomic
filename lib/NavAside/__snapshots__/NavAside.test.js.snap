// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPortal matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`NavAside matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`NavAside styles matches the snapshot 1`] = `
Object {
  "@global": Object {
    "body": Object {
      "overflow": "hidden",
    },
  },
  "@keyframes overlayFadeIn": Object {
    "from": Object {
      "opacity": 0,
    },
    "to": Object {
      "opacity": 1,
    },
  },
  "@keyframes showFromLeft": Object {
    "from": Object {
      "left": -320,
    },
    "to": Object {
      "left": 0,
    },
  },
  "@keyframes showFromRight": Object {
    "from": Object {
      "right": -320,
    },
    "to": Object {
      "right": 0,
    },
  },
  "block": Object {
    "backgroundColor": "#fff",
    "height": "100%",
    "maxWidth": "calc(100% - 32px)",
    "overflow": "hidden",
    "position": "fixed",
    "top": 0,
    "transition": "0.3s left, 0.3s right",
    "width": 320,
  },
  "card": Object {
    "cursor": "auto",
    "height": "100%",
    "overflow": "auto",
    "padding": 0,
  },
  "closeIcon": Object {
    "alignItems": "center",
    "display": "flex",
    "height": "32px",
  },
  "content": Object {
    "@media screen and (min-width:768px)": Object {
      "padding": "32px",
    },
    "backgroundColor": "#fff",
    "padding": "16px",
  },
  "hideLeft": Object {
    "left": -320,
  },
  "hideRight": Object {
    "right": -320,
  },
  "overlay": Object {
    "background": "rgba(0,0,0,0.8)",
    "cursor": "pointer",
    "height": "100%",
    "left": 0,
    "position": "fixed",
    "top": 0,
    "transition": "0.3s opacity",
    "width": "100%",
    "zIndex": 1000,
  },
  "overlayHide": Object {
    "opacity": 0,
  },
  "overlayShow": Object {
    "animation": "0.3s overlayFadeIn linear",
  },
  "showLeft": Object {
    "animation": "0.3s showFromLeft ease-out",
    "left": 0,
  },
  "showRight": Object {
    "animation": "0.3s showFromRight ease-out",
    "right": 0,
  },
  "top": Object {
    "@media screen and (min-width:768px)": Object {
      "padding": "32px",
    },
    "backgroundColor": "#F8FAFD",
    "padding": "16px",
  },
  "topContent": Object {
    "minWidth": 0,
  },
}
`;
